src/main/java/com/synopsys/integration/detect/lifecycle/boot/decision/AutoScanTypeDecider.java

```diff
@@ -0,0 +1,165 @@
+        while(mediaType != null) {
+            mediaTypes.addAll(mediaTypeRegistry.getAliases(mediaType));
+            mediaTypes.add(mediaType);
+            mediaType = mediaTypeRegistry.getSupertype(mediaType);
+        }
+        return mediaTypes.stream().noneMatch(x -> x.getType().equals("text"));
+    }
+    
+    private PathsCollection search(Path pathToSearch) {
+        PathsCollection pathsCollection = new PathsCollection();
+        long t1 = System.currentTimeMillis();
+        try {
+            Files.walkFileTree(pathToSearch, new FileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    if (isEligibleFile(file.toFile().getName())) {
+                        if (isBinary(file.toFile())) {
+                            pathsCollection.binaryPaths.add(file.toAbsolutePath().toString());
+                        } else if (pathsCollection.detectorPaths.isEmpty()) {
+                            pathsCollection.detectorPaths.add(pathToSearch.toAbsolutePath().toString());
+                        }
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+                
+                @Override
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                    String directoryName = dir.getFileName().toString().trim().toLowerCase();
+                    if(shouldAvoidDirectory(directoryName)){
+                        return FileVisitResult.SKIP_SUBTREE;
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+                
+                @Override
+                public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                    logger.error(exc.getMessage());
+                    return FileVisitResult.TERMINATE;
+                }
+                
+                @Override
+                public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+                    return FileVisitResult.CONTINUE;
+                }
+                
+            });
+            pathsCollection.signaturePaths.add(pathToSearch.toAbsolutePath().toString());
+        } catch (IOException ex) {
+            logger.error("Failure when attempting to locate build config files.", ex);
+        } finally {
+            logger.debug("Done. Seconds: {}", (System.currentTimeMillis()-t1)/1000D);
+        }
+        return pathsCollection;
+    }
+    
+    class PathsCollection {
+        private final Set<String> binaryPaths = new HashSet<>();
+        private final Set<String> detectorPaths = new HashSet<>(); 
+        private final Set<String> signaturePaths = new HashSet<>();
+    }
```

Comment:
Move PathsCollection to separate class file?

Reply:
Fine where it is.